### DISCUSSION TITLE
PGP send

### DISCUSSION THREAD

**Discussion by asterysk** [simple-login/app#467] [Q&A]
I have managed to use PGP to receive forwarded emails via simplelogin to protonmail.

Have been experimenting with how to best incorporate PGP into the sending also and it is proving very complicated, so if I can start with one aspect :

Is there a way to enable PGP encryption on the Protonmail (PM) send to SimpleLogin (SL) i.e. just the hop from PM to SL so that when SL forwards to the recipient it goes onwards in plaintext. 

I tried adding the SL public key to my reverse alias contact in PM and this resulted in the recipient getting a PGP encrypted email which implies that SL is not decrypting the incoming mail from PM

**Comment by nguyenkims** [comment]
Hey do you think this feature request in https://github.com/simple-login/app/discussions/429 can correspond to the feature you describe?

**Reply by bcronce** [reply]
They do sound possibly the same, but this request sounds more correct. The other request is asking to add private keys, which isn't something that should be done. Using SL's public key and letting SL decrypt with their private key is the better way than setting up your own key pair. It makes it very clear than the encryption is between you and SL.

Then again, perfect is the enemy of good. Using SL as a PGP proxy could be a benefit for power users that properly understand their threat model.

**Reply by nguyenkims** [reply]
Oh I see, do you think the below can work? 

- SL provides a "catch-all" PGP public key for all reverse-aliases. 
- When sending an email to a reverse-alias, user can choose to encrypt the email with that key 
- if that's the case, SimpleLogin will first decrypt the email, then send the email to the contact, eventually encrypts the email again if the contact has PGP enabled.

**Reply by bcronce** [reply]
I just want to make it clear that I did not write the original request, but I am trying to help interpret it. That out of the way.

Your steps do look to make sense.

When I originally replied seeing that this was a few week old, I was thinking strictly from a public/private key system and not how PGP uses keys associated to email addresses. Not having used PGP in a while, I cannot think of how PGP email integration might interact with a "catch all". I'm assuming if it's the same signature, regardless of the actual target email, this would work. But I could see some plugin possibly trying to "help" by associating the destination address with the public key and not finding it.

Fundamentally there should be no issue that I, some person on the internet, can think of.

**Reply by jeremygaither** [reply]
I think these are two separate requests. The Trello item seems to create a pgp encrypted message to the reverse alias contact using a distinct key, where SimpleLogin encrypts a plaintext message to the recipient, but I read this request as allowing ProtonMail->simplelogin pgp encrypted messages that are unencrypted before being sent to the reverse alias recipient. I can see both use cases as valid. For this request, to decrypt prior to sending, a catch-all PHP private key (per SL User account?) generated by SL might work. I would prefer an optional unique private key per alias, still SL generated), so each ProtonMail->SimpleMail email has its own key pair and metadata. In the case a pgp key for encrypted mail to the alias recipient was added and source->SL pgp encryption was used )both are true), simplelogin would play the part of useful man in the middle (usually an attack). Pass-through crypto could fingerprint SL aliases to specific ProtonMail accounts or PGP keys, and I think that would defeat the purpose of more privacy with ProtonMail and SimpleLogin. But if the intent is to both encrypt the outbound reply traffic by using an alias specific key, and secure reply communications to SL from a pgp-encrypted source, that could sanitize some metadata but not all (provided that the recipient name wasn't in the original outgoing email, and somehow the subject was munged).

**Reply by ataraskov** [reply]
> Oh I see, do you think the below can work?
> 
> * SL provides a "catch-all" PGP public key for all reverse-aliases.
> * When sending an email to a reverse-alias, user can choose to encrypt the email with that key
> * if that's the case, SimpleLogin will first decrypt the email, then send the email to the contact, eventually encrypts the email again if the contact has PGP enabled.

i like this approach

**Reply by nathanstill** [reply]
I feel like this could be a simple win.  The problem is even if PGP is on for receiving, as soon as an email is sent out, the email exists in an unencrypted form as a sent item.

**Reply by StummeJ** [reply]
Also looking for this feature

**Reply by Xynonners** [reply]
> > Oh I see, do you think the below can work?
> > 
> > * SL provides a "catch-all" PGP public key for all reverse-aliases.
> > * When sending an email to a reverse-alias, user can choose to encrypt the email with that key
> > * if that's the case, SimpleLogin will first decrypt the email, then send the email to the contact, eventually encrypts the email again if the contact has PGP enabled.
> 
> i like this approach

anonaddy implemented this approach.
